crates/node/src/lib.rs

```diff
@@ -923,6 +955,20 @@ impl Node {
             return Ok(());
         };
 
+        let application_version = self.ctx_manager.get_application_latest_version(
+            &meta_value.application_id.clone().into_string().into(),
+        )?;
+
+        let message = serde_json::to_vec(&types::CatchupStreamMessage::ResponseMeta(
+            types::CatchupResponseMeta {
+                application_id: meta_value.application_id.into_string().into(),
+                version: application_version,
+            },
+        ))?;
+        stream
+            .send(calimero_network::stream::Message { data: message })
+            .await?;
```

Comment:
so long as it's guaranteed that we only send `ResponseMeta` when there's a hash variance

so, we can add a check above this that responds with a variant for "yah, you're good" or "you're up to date"

Reply:
Ideally all synchronizable activity including things like application upgrade should be a transaction